### ISSUE TITLE
Button w/o label (i.e. icon only button) should have a11yTitle

### ISSUE THREAD

**Issue by taysea** [grommet/eslint-plugin-grommet#14]
**Please describe what the rule should do:**

Flag if Button without label with icon prop (no children) doesn't have `a11yTitle`. Otherwise, non-specific title will be read to screen reader users.

**What category of rule is this? (place an "X" next to just one item)**

[x] Warns about a potential error (problem)
[ ] Suggests an alternate way of doing something (suggestion)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule will warn about:**

```js
<Button icon={<FormClose />} />
```

**Why should this rule be included in eslint-plugin-grommet?**

Accessibility best practice.
